### PR TITLE
test: add a test case for transitive dependency

### PR DESCRIPTION
**Pull request by hi-rustin**
ref https://github.com/hi-rustin/cargo-information/issues/29

Currently, we always select the highest version.

This PR just adds a test case to cover the current behavior. We will discuss it in the issue.

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check (PR reviewed by myself and ready for feedback.)

**Comment by hi-rustin** [comment]
@epage ping~

File: tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    // Dep1 depends on 3.0.0, Dep2 depends on 2.0.0, Dep3 depends on 1.0.0
+    cargo_test_support::registry::Package::new("dep1", "1.0.0")
+        .dep("my-package", "3.0.0")
+        .publish();
+    cargo_test_support::registry::Package::new("dep2", "1.0.0")
+        .dep("my-package", "2.0.0")
+        .publish();
+    cargo_test_support::registry::Package::new("dep3", "1.0.0")
+        .dep("my-package", "1.0.0")
+        .publish();
```
Comment (epage) [inline tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs:21]: nit: Might make it easier to read if you align `dep<N>` with `<N>.0.0`

File: tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        .assert()
+        .stdout_matches(file!["stdout_project_root.log"])
+        .stderr_matches(file!["stderr_project_root.log"]);
+    cargo_info()
+        .arg("my-package")
+        .arg("--registry=dummy-registry")
+        .current_dir(crate1_directory)
+        .assert()
+        .stdout_matches(file!["stdout_crate1.log"])
+        .stderr_matches(file!["stderr_crate1.log"]);
```
Comment (epage) [inline tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs:44]: imo I'd name these `crate1.stdout.log`, etc so they are grouped by role in the tests

**Review by epage — APPROVED** [review]

File: tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs (hunk 1 above)
Comment (hi-rustin) [reply tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs]: Updated.

File: tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs (hunk 2 above)
Comment (hi-rustin) [reply tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs]: Updated.

File: tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs (hunk 1 above)
Comment (epage) [reply tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs]: Sorry, I meant with `my-package` version requirement

File: tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs (hunk 2 above)
Comment (epage) [reply tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs]: `stdout_project_root` wasn't updated

**Comment by epage** [comment]
FYI I'm most of the way there on an implementation for #29 that I'll rebase onto this test once its ready

File: tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs (hunk 1 above)
Comment (hi-rustin) [reply tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs]: Thank you!

File: tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs (hunk 2 above)
Comment (hi-rustin) [reply tests/testsuite/cargo_information/transitive_dependency_within_ws/mod.rs]: Good cacth!

**Pull request merged**